tests/Oryx.Tests.Common/EndToEndTestHelper.cs

```diff
@@ -255,6 +221,12 @@ public static class EndToEndTestHelper
                 {
                     // Stop the container so that shared resources (like ports) are disposed.
                     dockerCli.StopContainer(runResult.ContainerName);
+
+                    // Access the output and error streams after the process has exited
```

Comment:
This solution is different from what I mentioned to you yesterday because even in the scenario where the caller supplies the builders for stdout or stderr streams, the caller still needs to read the builders only after the process no longer writes to them to prevent the exception that we are seeing, so instead of changing the method contract, I just changed this method to make sure to read the output at a point which is safe.